core/src/blocks/data_source.rs

```diff
@@ -75,18 +75,14 @@ impl DataSource {
     async fn search_data_source(
         &self,
         env: &Env,
-        workspace_id: Option<String>,
+        workspace_id: String,
         data_source_id: String,
         top_k: usize,
         filter: Option<SearchFilter>,
         target_document_tokens: Option<usize>,
     ) -> Result<Vec<Document>> {
-        let data_source_project = match workspace_id {
-            Some(workspace_id) => {
-                get_data_source_project(&workspace_id, &data_source_id, env).await?
-            }
-            None => env.project.clone(),
```

Comment:
This used to be possible through the dust command line tool so should not break any integration